File: generate_favicon.py
import cairosvg
from PIL import Image
import os

svg_path = 'public/favicon.svg'
png_path = 'public/favicon_temp.png'
ico_path = 'app/favicon.ico'

# Convert SVG to PNG
cairosvg.svg2png(url=svg_path, write_to=png_path, output_width=512, output_height=512)

# Convert PNG to ICO
img = Image.open(png_path)
img.save(ico_path, format='ICO', sizes=[(16, 16), (32, 32), (48, 48), (64, 64), (128, 128), (256, 256)])
print(f"Successfully generated {ico_path}")

# Optional: Generate one in public/ as well for hardcoded links
img.save('public/favicon.ico', format='ICO', sizes=[(16, 16), (32, 32), (48, 48), (64, 64), (128, 128), (256, 256)])

# Clean up
os.remove(png_path)

File: generate-favicon.js
const fs = require('fs');
const path = require('path');
const svg2img = require('svg2img');
const pngToIco = require('png-to-ico');

const svgPath = path.join(__dirname, 'public', 'favicon.svg');
const pngPath = path.join(__dirname, 'public', 'favicon-temp.png');
const icoPath = path.join(__dirname, 'app', 'favicon.ico');
const publicIcoPath = path.join(__dirname, 'public', 'favicon.ico');

svg2img(svgPath, { width: 512, height: 512 }, (error, buffer) => {
    if (error) {
        console.error('Error generating PNG:', error);
        process.exit(1);
    }
    fs.writeFileSync(pngPath, buffer);
    console.log('PNG generated at', pngPath);

    pngToIco(pngPath).then(buf => {
        fs.writeFileSync(icoPath, buf);
        fs.writeFileSync(publicIcoPath, buf);
        console.log('ICO generated at', icoPath, 'and', publicIcoPath);
        fs.unlinkSync(pngPath);
    }).catch(console.error);
});
